src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -1110,6 +1144,15 @@ public Settings additionalSettings() {
         return builder.build();
     }
 
+    @Override
```

Comment:
So the way Craig set this up, the idea is to register a set of extension security settings when an extension is installed/registered as part of the core initialization process. To make this work with service accounts the general idea would be to make sure that a service account is only able to operate on the indices listed in the reserved_indices list and affiliated with themselves--this should use a mapping or some other mechanism.

Reply:
I have removed BACKEND_ROLES setting for now. In SecurityIndexAccessEvaluator we check by looking up extension settings by username (should match extension name), and then matching the requested index with extension's reserved index to determine whether access should be granted